alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt

```diff
@@ -69,47 +71,82 @@ class DocLevelMonitorQueries(private val client: AdminClient, private val cluste
-                                request.add(indexRequest)
-                            }
-
-                            queryClient.bulk(request, docLevelQueryIndexListener)
-                        }
 
-                        override fun onFailure(e: Exception) {
-                            if (indexMonitorActionListener != null) {
-                                indexMonitorActionListener.onFailure(AlertingException.wrap(e))
-                            } else executeMonitorActionListener?.onFailure(AlertingException.wrap(e))
+        val getAliasesRequest = GetAliasesRequest(index)
+        queryClient.admin().indices().getAliases(
+            getAliasesRequest,
+            object : ActionListener<GetAliasesResponse> {
+                override fun onResponse(getAliasesResponse: GetAliasesResponse?) {
+                    val aliasIndices = getAliasesResponse?.aliases?.keys()?.map { it.value }
+                    val isAlias = aliasIndices != null && aliasIndices.isNotEmpty()
+                    val indices = if (isAlias) getAliasesResponse?.aliases?.keys()?.map { it.value } else listOf(index)
+                    val indexRequests = mutableListOf<IndexRequest>()
+                    log.info("indices: $indices")
+                    indices?.forEach { indexName ->
+                        if (clusterState.routingTable.hasIndex(indexName)) {
+                            val indexMetadata = clusterState.metadata.index(indexName)
+
+                            if (indexMetadata.mapping() != null) {
+                                log.info("Index name: $indexName")
+                                val properties = (
+                                    (indexMetadata.mapping()?.sourceAsMap?.get("properties"))
+                                        as Map<String, Map<String, Any>>
+                                    )
+                                val updatedProperties = properties.entries.associate { "${it.key}_$monitorId" to it.value }.toMutableMap()
+
+                                val updateMappingRequest = PutMappingRequest(ScheduledJob.DOC_LEVEL_QUERIES_INDEX)
+                                updateMappingRequest.source(mapOf<String, Any>("properties" to updatedProperties))
+
+                                queryClient.admin().indices().putMapping(
+                                    updateMappingRequest,
+                                    object : ActionListener<AcknowledgedResponse> {
+                                        override fun onResponse(response: AcknowledgedResponse) {
+
+                                            queries.forEach {
+                                                var query = it.query
+
+                                                properties.forEach { prop ->
+                                                    query = query.replace("${prop.key}:", "${prop.key}_$monitorId:")
+                                                }
```

Comment:
Should query property now also have the index name now? Given, now queries in the query document will be repeated per index, do we need an additional filtering along with the monitor_id?

Reply:
Yes that has been added from Subho's comment. I think you are not seeing the latest commit

Reply:
Just saw the update, thanks. I hope we are considering this during index creation, update time as well.